Header header
int8[] ports
int8[] states
